zcash_client_backend/src/data_api/testing/transparent.rs

```diff
@@ -243,5 +279,90 @@ where
+}
+
+/// This test attempts to verify that transparent funds spendability is
+/// accounted for properly given the different minimum confirmations values
+/// that can be set when querying for balances.
+pub fn transparent_balance_spendability<DSF>(dsf: DSF, cache: impl TestCache)
+where
+    DSF: DataStoreFactory,
+{
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(dsf)
+        .with_block_cache(cache)
+        .with_account_from_sapling_activation(BlockHash([0; 32]))
+        .build();
+
+    let account = st.test_account().cloned().unwrap();
+    let uaddr = st
+        .wallet()
+        .get_current_address(account.id())
+        .unwrap()
+        .unwrap();
+    let taddr = uaddr.transparent().unwrap();
+
+    // Initialize the wallet with chain data that has no shielded notes for us.
+    let not_our_key = ExtendedSpendingKey::master(&[]).to_diversifiable_full_viewing_key();
+    let not_our_value = NonNegativeAmount::const_from_u64(10000);
+    let (start_height, _, _) =
+        st.generate_next_block(&not_our_key, AddressType::DefaultExternal, not_our_value);
+    for _ in 1..10 {
+        st.generate_next_block(&not_our_key, AddressType::DefaultExternal, not_our_value);
+    }
+    st.scan_cached_blocks(start_height, 10);
+
+    // The wallet starts out with zero balance.
+    check_balance::<DSF>(
+        &st as &TestState<_, DSF::DataStore, _>,
+        &account,
+        taddr,
+        0,
+        &Balance::ZERO,
+    );
+    // we currently treat min_confirmation the same regardless they are 0 or 1
+    check_balance::<DSF>(&st, &account, taddr, 1, &Balance::ZERO);
+
+    // Create a fake transparent output.
+    let value = NonNegativeAmount::from_u64(100000).unwrap();
+    let txout = TxOut {
+        value,
+        script_pubkey: taddr.script(),
+    };
+
+    // Pretend the output was received in the chain tip.
+    let height = st.wallet().chain_height().unwrap().unwrap();
+    let utxo = WalletTransparentOutput::from_parts(OutPoint::fake(), txout, Some(height)).unwrap();
+    st.wallet_mut()
+        .put_received_transparent_utxo(&utxo)
+        .unwrap();
+
+    // The wallet should detect the balance as available
+    let mut zero_or_one_conf_value = Balance::ZERO;
+
+    // add the spendable value to the expected balance
+    zero_or_one_conf_value.add_spendable_value(value).unwrap();
+
+    check_balance::<DSF>(&st, &account, taddr, 0, &zero_or_one_conf_value);
+    // we currently treat min_confirmation the same regardless they are 0 or 1
+    check_balance::<DSF>(&st, &account, taddr, 1, &zero_or_one_conf_value);
+
+    // now if we increase the number of confirmations our spendable balance should
+    // be zero and the total balance equal to `value`
+    let mut not_confirmed_yet_value = Balance::ZERO;
+
+    not_confirmed_yet_value
+        .add_pending_spendable_value(value)
+        .unwrap();
+
+    check_balance::<DSF>(&st, &account, taddr, 2, &not_confirmed_yet_value);
```

Comment:
In the current test failure, the spendable value at the mempool height is being calculated at 100000 zats here, but the test expects it to be 0 (because the call on line 358 is to `add_pending_spendable_value`, not `add_spendable_value`). The test looks correct to me: the funds should indeed not be spendable at two confirmations.

Reply:
oh. I'll look into this!

Reply:
This is interesting. I ran `cargo test --release --workspace --features ' bundled-prover download-params lightwalletd-tonic sync temporary-zcashd transparent-inputs unstable unstable-serialization unstable-spanning-tree  test-dependencies '` in MacOs and the failing test does not appear to be ran.

Reply:
uuhmmm Interesting

Local on MacOS
```
test wallet::transparent::tests::transparent_balance_across_shielding ... ok
test wallet::transparent::tests::transparent_balance_spendability ... ok
```

Test on Linux CI
```
test wallet::transparent::tests::transparent_balance_across_shielding ... ok
test wallet::transparent::tests::transparent_balance_spendability ... FAILED
```

Reply:
fixed that ... I'm a prisoner of my own copypasta